lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -727,72 +827,175 @@ def _labeled_unit_config(
 
         Args:
             charm: a `CharmBase` object that manages this
                 `MetricsEndpointProvider` object. Typically this is
                 `self` in the instantiating class.
-            name: a string name of the relation between `charm` and
-                the Prometheus charmed service.
-            service_event: a `CharmEvent` in response to which each charm unit
-                must advertise its scrape endpoint host address.
+            relation_name: an optional string name of the relation between `charm`
+                and the Prometheus charmed service. You will need to specify this
+                parameter only if your charm provides more than one relation with
+                `prometheus_scrape` as interface, which is very uncommon: if your
+                charm has multiple metrics endpoints, you can expose them over
+                one relation as separate entries in the `jobs` list.
+                If `relation_name` is not specified, the
+                `MetricsEndpointProvider` will look for the existence of precisely
+                one relation provided by the charm and using the `prometheus_scrape`
+                interface and, if none or multiple are found, it will raise a
+                ops.model.ModelError exception.
             jobs: an optional list of dictionaries where each
                 dictionary represents the Prometheus scrape
                 configuration for a single job. When not provided, a
                 default scrape configuration is provided for the
-                `/metrics` endpoint pooling using port `80`.
+                `/metrics` endpoint polling all units of the charm on port `80`
+                using the `MetricsEndpointProvider` object.
             alert_rules_path: an optional path for the location of alert rules
-                files.  Defaults to "src/prometheus_alert_rules" at the top level
-                of the charm repository.
+                files.  Defaults to "./prometheus_alert_rules",
+                resolved from the directory hosting the charm entry file.
+                The alert rules are automatically update on charm upgrade.
         """
-        super().__init__(charm, name)
+        if not relation_name:
+            try:
+                # Check if there is just one relation with the right interface
+                relation_name = _get_single_relation_by_interface(
+                    charm, RELATION_INTERFACE_NAME, RelationDirection.PROVIDED
+                )
+            except NoRelationWithInterfaceFoundError:
+                raise ModelError(
+                    f"No provided relation with the '{RELATION_INTERFACE_NAME}' interface found; "
+                    f"did you add a relation with the '{RELATION_INTERFACE_NAME}' interface in "
+                    "the charm's metadata.yaml file?"
+                )
+            except MultipleRelationsWithInterfaceFoundError as e:
```

Comment:
Using exception messages for communicating to the developer information that ideally should have been statically inferred reminded me of "explicit is better than implicit": looking at all the changes that were necessary to have the relation an optional argument do not seem to be worth it.